crates/compiler/test_gen/src/gen_str.rs

```diff
@@ -436,7 +520,17 @@ fn small_str_concat_empty_second_arg() {
 }
 
 #[test]
-#[cfg(any(feature = "gen-llvm", feature = "gen-dev"))]
+#[cfg(feature = "gen-wasm")]
+fn small_str_concat_empty_second_arg() {
+    assert_evals_to!(
+        r#"Str.concat "01234567890" """#,
+        [0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30, 0x8b],
+        [u8; 12]
+    );
+}
```

Comment:
@bhansconnect not every test was merged or copied, for instance I'm guessing tests like this which work on `[u8; 12]` as opposed to the same test for `gen-llvm` works with `[u8; 24]` are the reason for the original split. There's only like 4 test which need alternative forms for `gen-wasm`, so I think it's a win to merge everything into one `gen_str.rs`. There was a small drift between what was being tested on `gen-wasm` vs `gen-llvm`/`gen-dev` because people added tests to one place and not the other and putting them in one place will hopefully keep test coverage in sync.